### ISSUE TITLE
`ToolbarButtonComponent` should allow to pass aria attributes

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#18037]
## Description

`ToolbarButtonComponent` does not allow to pass aria attributes such a `aria-haspopup`, `aria-expanded`, or `aria-controls` and thus cannot be used as a button which controls a pop-up menu. Possibly other attributes should be allowed too.

`aria-controls` (and `ariaControlsElements` https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaControlsElements) would be also useful for the debugger icon button in the toolbar as it controls expansion of  the debugger sidebar.

## Reproduce

https://github.com/jupyterlab/jupyterlab/blob/f37154e23ae423bfb77d6e6b7aee33eb087f7fcf/packages/ui-components/src/components/toolbar.tsx#L767-L803

## Expected behavior

Allows to implement https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/

## Context

- JupyterLab version: 4.5.0b1 and earlier

**Comment by agriyakhetarpal** [comment]
Hi! @krassowski and I have been discussing this internally at work, and I believe this could be a valuable first contribution to JupyterLab. Since it looks like it's going to be a self-contained change, I would like to take this on! I'll put together a PR shortly.

**Comment by krassowski** [comment]
There are two ways to approach it:
- include everything with `React.AriaAttributes` typing: we may end up allowing things which do not make sense on buttons
- curate allowed keys by using `Pick<>` to only include relevant aria attributes; this is closer to what https://stevekinney.com/courses/react-typescript/accessibility-and-aria-typing arguments for, at a disadvantage that future additions will not be picked up automatically
